app/scripts/lib/metrics.js

```diff
@@ -121,6 +121,7 @@ define(function (require, exports, module) {
 
     this._marketingImpressions = {};
     this._activeExperiments = {};
+    this._eventMemory = [];
```

Comment:
Like @rfk suggested, lookups could be a bit quicker by using an Object instead of an Array, though I don't imagine the list ever growing so long that lookups become a problem. 

Another alternative is to skip storing a list of events locally and search this.events directly:

```js
isEventLogged (eventName) {
  return _.some(this.events, (event) => event.type === eventName);
}
```